src/canvas/canvas.controller.ts

```diff
@@ -221,11 +229,47 @@ export class CanvasController {
    * ```
    *
    * @param {ListFilter} filter - The filter to apply when retrieving items.
+   * @param req - HTTP request object
    * @return {Promise<any>} - A Promise that resolves to the retrieved items.
    */
   @Get('/')
   @UseGuards(AuthenticatedGuard)
-  public async readAll(@Query() filter: ListFilter): Promise<any> {
-    return this.canvasService.readAll(filter);
+  public async readAll(
+    @Query() filter: ListFilter,
+    @Req() req: Request,
+  ): Promise<any> {
+    return await this.canvasService.readAll(filter, req.user.toString());
+  }
+
+  /**
+   * Gives access to a single canvas for a single user
+   * @param canvasId
+   * @param dto - an object containing 'userId'
+   */
+  @Post('/:id/access')
+  @UseGuards(AuthenticatedGuard, CanvasGuard)
+  public async giveAccess(
+    @Param('id') canvasId: Uuid,
+    @Body() dto: CanvasAccessDTO,
+  ): Promise<any> {
+    const userId = dto.userId;
+    await this.canvasService.giveAccess({ canvasId, userId });
+    return [];
```

Comment:
jbc. to nie trzeba zawsze czegoś zwracać. Można usunąć `: Promise<any>` i `return [];` i efekt będzie ten sam 😀